Automattic/wp-calypso
# Marketplace Theme Install: Fix stuck loading screen (take 2)

When a user adds a Partner theme during onboarding (Design Picker) and then removes the Partner theme from the checkout and purchases only the Business plan, the user is redirected to /marketplace/thank-you page.

Since the theme was removed, the page loads indefinitely for the Atomic transfer and the theme installation to finish - thus, blocking the user on this page.



Related to https://github.com/Automattic/wp-calypso/issues/96864

## Proposed Changes

* Check if the user has actually purchased the theme
* If not bail and redirect to /home

## Why are these changes being made?


* if the users selected a Partner theme during onboarding and then removed it from checkout, the Marketplace interface will remain blocked in the loading state

## Testing Instructions



* Use the live link
* Go to /start and create a new site (free)
* Go to Design Picker and select a Partner theme (SolarOne)
* In checkout, remove the SolarOne product
* Purchase only the Business
* You're redirected to /marketplace/thank-you
* You're redirected to /home/
* Open /themes in incognito
* Copy the with-theme signup url for a premium/partner theme
* Create a site with the theme
* Make sure it works
* Note: For premium themes we are still displaying the congrats screen

## Pre-merge Checklist
